src/main/java/uk/gov/hmcts/reform/prl/controllers/C100IssueCaseController.java

```diff
@@ -55,30 +59,41 @@ public AboutToStartOrSubmitCallbackResponse issueAndSendToLocalCourt(
             requireNonNull(caseData);
             sendgridService.sendEmail(c100JsonMapper.map(caseData));
         }
+        caseData = caseData.toBuilder().issueDate(LocalDate.now()).build();
         Map<String, Object> caseDataUpdated = callbackRequest.getCaseDetails().getData();
 
         if (null != caseData.getCourtList() && null != caseData.getCourtList().getValue()) {
-            String baseLocationId = caseData.getCourtList().getValue().getCode();
+            String[] idEmail = caseData.getCourtList().getValue().getCode().split(":");
+            String baseLocationId = Arrays.stream(idEmail).toArray()[0].toString();
             String key = locationRefDataService.getCourtDetailsFromEpimmsId(baseLocationId, authorisation);
             String[] venueDetails = key.split("-");
             String regionId = Arrays.stream(venueDetails).toArray()[1].toString();
             String courtName = Arrays.stream(venueDetails).toArray()[2].toString();
             String regionName = Arrays.stream(venueDetails).toArray()[4].toString();
             String baseLocationName = Arrays.stream(venueDetails).toArray()[5].toString();
+            caseDataUpdated.put("courtName", courtName);
             caseDataUpdated.put("caseManagementLocation", CaseManagementLocation.builder()
                 .regionId(regionId).baseLocationId(baseLocationId).regionName(regionName)
                 .baseLocationName(baseLocationName).build());
-            caseData = caseData.toBuilder().issueDate(LocalDate.now()).courtName(courtName).build();
+            String courtEmail = Arrays.stream(idEmail).toArray()[1].toString();
+            caseDataUpdated.put("localCourtAdmin", List.of(Element.<LocalCourtAdminEmail>builder().id(UUID.randomUUID())
+                                                               .value(LocalCourtAdminEmail.builder().email(courtEmail)
+                                                                          .build()).build()));
+            caseData.setCourtName(caseDataUpdated.get("courtName").toString());
         }
-
+        caseData.setIssueDate();
         // Generate All Docs and set to casedataupdated.
         caseDataUpdated.putAll(documentGenService.generateDocuments(authorisation, caseData));
-
         // Refreshing the page in the same event. Hence no external event call needed.
         // Getting the tab fields and add it to the casedetails..
         Map<String, Object> allTabsFields = allTabsService.getAllTabsFields(caseData);
+        caseDataUpdated.put("issueDate", LocalDate.now());
```

Comment:
there is a reason for the old code. its about return and resubmit. is this change valid?